connectors/jira/src/test/java/com/vmware/connectors/jira/JiraControllerTests.java

```diff
@@ -73,6 +76,42 @@ public void testDiscovery() throws Exception {
         testConnectorDiscovery();
     }
 
+    @Test
+    public void testRegex() throws Exception {
+        List<String> expected = ImmutableList.of(
+                "ABC-1",
+                "ABCDEFGHIJ-123",
+//                "ABCDEFGHIJK-456", // Should not match because too many letters
+                "ABC-3", // not ideal, but not worth fixing
+                "ABC-4",
+                "ABC-5",
+//                "ABC-6", // Should not match due to leading "x"
+                "ABC-7",
+//                "ABC-8", // Should not match due to leading "1"
+                "ABC-9",
+                "ABC-10",
+                "ABC-11",
+                "ABC-12", // not ideal, but not worth fixing
+                "ABC-13",
+                "ABC-14",
+                "ABC-15", // not ideal, but not worth fixing
+                "ABC-16",
+                "ABC-17",
+                "ABC-18",
+                "ABC-19", // not ideal, but not worth fixing
+//                "ABC-20", // Should not match due to leading "x"
+//                "D-2", // should not match
```

Comment:
This D-2 through SA-2 is what the false positives were that were reported.  SA-2 was moved to APPSEC-2 (which made the false positive more confusing -- we didn't see APPSEC-2 in the email).

I started off with the false positives in the results (to make sure it was behaving as expected), then fixed the code and test, but left the commented out false positives to document what we're acknowledging the tradeoffs are.